src/entry.c

```diff
@@ -39,8 +41,18 @@
 static void xcb_xrm_append_char(xcb_xrm_entry_t *entry, xcb_xrm_entry_parser_state_t *state,
         const char str) {
     if (state->buffer_pos == NULL) {
-        memset(&(state->buffer), 0, sizeof(state->buffer));
+        FREE(state->buffer);
+        state->buffer = calloc(1, BUFFER_SIZE);
         state->buffer_pos = state->buffer;
+        if (state->buffer == NULL) {
+            /* Let's ignore this character and try again next time. */
+            return;
+        }
+    }
+
+    /* Increase the buffer if necessary. */
+    if ((state->buffer_pos - state->buffer) % BUFFER_SIZE == BUFFER_SIZE - 1) {
+        state->buffer = realloc(state->buffer, (state->buffer_pos - state->buffer) + BUFFER_SIZE + 1);
```

Comment:
Right. TIL that the difference between two pointers is only defined if they both point in the same object and is then of type `ptrdiff_t`. I've updated the PR.